cubeviz/tools/collapse_cube.py

```diff
@@ -311,187 +228,321 @@ def hide_start_end(self, dohide):
+            self.ui.error_label.setText('No end index, assuming last spectral wavelength.')
+            end_index = len(self.wavelengths) - 1
+
+        try:
+            start_index = int(start_index)
+        except:
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start index is not an integer.')
+            self.ui.error_label.setVisible(True)
+
+            start_index = None
+
+        try:
+            end_index = int(end_index)
+        except:
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('End index is not an integer.')
+            self.ui.error_label.setVisible(True)
+
+            end_index = None
+
+        if start_index < 0:
+            self.ui.start_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('Start index is out of range, setting to end point.')
+            self.ui.error_label.setVisible(True)
+
+            start_index = 0
+            
+        if start_index > len(self.wavelengths):
+            self.ui.start_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start index is out of range.')
+            self.ui.error_label.setVisible(True)
+
+            start_index = None
+            
+
+        if end_index > len(self.wavelengths):
+            self.ui.end_label.setStyleSheet("color: rgba(0, 0, 255, 128)")
+            self.ui.error_label.setText('End index is out of range, setting to end point.')
+            self.ui.error_label.setVisible(True)
+
+            end_index = len(self.wavelengths)-1
+           
+        if end_index < 0:
+            self.ui.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('End index is out of range.')
+            self.ui.error_label.setVisible(True)
+
+            end_index = None
+
+        if end_index < start_index:
+            self.ui.end_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Start index must be less than the end index.')
+            self.ui.error_label.setVisible(True)
+            return None, None
+
+        log.debug('  returning with start_index {} and end_index {}'.format(
+            start_index, end_index))
+
+        return start_index, end_index
+
+    def _calculate_callback_simple_sigma_check(self):
+        log.debug('_calculate_callback_simgple_sigma_check')
+
+        simple_sigma = self.ui.simple_sigma_input.text().strip()
+        log.debug('    simple_sigma {}'.format(simple_sigma))
+
+        try:
+            simple_sigma = float(simple_sigma)
+        except:
+            self.ui.simple_sigma_label.setStyleSheet("color: rgba(255, 0, 0, 128)")
+            self.ui.error_label.setText('Sigma must be a floating point number.')
+            self.ui.error_label.setVisible(True)
+            return None
+
```

Comment:
Is it okay to allow `sigma <= 0` here?

Reply:
I added a check here.

Reply:
Now if I put in a value for sigma, but not for Start Wavelength or End Wavelength, then the "End Wavelength" label turns purple but nothing else happens. Shouldn't I be able to apply sigma clipping to the entire cube by default without explicitly specifying start and end points?

Reply:
added a message in the final dialog to show the start and end wavelengths were used. removed the purple thing.